components/brave_wallet_ui/page/screens/fund-wallet/deposit-funds.tsx

```diff
@@ -550,6 +559,18 @@ function DepositAccount() {
     [copyAddressToClipboard]
   )
 
+  React.useEffect(() => {
+    ; (async () => {
+      if (!selectedAccount) {
+        return
+      }
+      const address = await generateReceiveAddress(
+        selectedAccount.accountId
+      ).unwrap()
+      setReceiveAddress(address)
+    })()
+  }, [selectedAccount])
```

Comment:
This can be avoided if we change `generateReceiveAddress` to a query. You can force the query to always fetch (ignore cache) with the `refetchOnMountOrArgChange: true` option

Reply:
This technically mutates backend and may produce different results for two consecutive calls. Applies for BTC and will be for ZEC